### ISSUE TITLE
Consider removing dict baseclass

### ISSUE THREAD

**Issue by nickstanisha** [nickstanisha/nesteddict#5]
Because of `NestedDict`'s behavior when indexing tuples, the following situation could arise for a user of this project
```python
def my_user_function(dictobject):
    assert(isinstance(dictobject, dict)), "I'm a function that only works on dicts"
    # start indexing tuples into dictobject
    return dictobject

dictobject == nesteddictobject  # True
my_user_function(dictobject) == my_user_function(nesteddictobject)  # False
```
In response to this dilemma, I could either
* Do nothing -- relying on the user to distinguish use-cases using `isinstance(obj, NestedDict)`
* Use an internal `self._dict` object instead of subclassing `dict`

Using approach 2, the following conditions would have to be satisfied
1) `isinstance(obj, dict) == False`
2) Attributes not found in `NestedDict` will default to attributes in the internal `self._dict` (effectively replicating the current class inheritance)
    * `getattr(NestedDict, attr)` should _never_ retrieve the default attrs that `NestedDict` gets by virtue of being a Python object (e.g. `__eq__`, `__contains__`, etc.). The available attributes should _only_ reflect what is hard-coded in `nesteddict.py` and what's present in `dir(dict())`
3) `import json; json.dumps(nesteddict)` still works (nice to have)
4) `dict(nesteddict)` and `NestedDict(dict)` still work

**Comment by nickstanisha** [comment]
@meowklaski Feel free to develop and make pull requests against the `no_inheritance` branch. I've already added the minimum level of functionality to pass all of the tests, but I'm not satisfied with it yet because it doesn't fully satisfy point 2 above.

Once the `no_inheritance` branch is ready, I'll do a final assessment of whether or not the additional code complexity is worth avoiding the headache in the code sample above.
